sktime/transformations/series/subsequence_extraction.py

```diff
@@ -0,0 +1,194 @@
+        -------
+        self : reference to self
+        """
+        if self.subseq_len > len(X):
+            raise ValueError(
+                f"Subsequence length parameter ({self.subseq_len}) is not less \
+                than or equal to the minimum sequence length of X ({len(X)})."
+            )
+
+        if not (isinstance(self.aggregate_fn, _ArrayFunctionDispatcher)):
+            raise ValueError(
+                f"{self.aggregate_fn} is not supported for parameter aggregate"
+            )
+
+        if self.selector not in ["max", "min"]:
+            raise ValueError(f"{self.selector} is not supported for parameter selector")
+
+        return self
+
+    def _transform(self, X, y=None):
+        """Transform X and return a transformed version.
+
+        private _transform containing core logic, called from transform
+
+        Parameters
+        ----------
+        X : nested pandas DataFrame of shape [n_instances, n_features]
+            each cell of X must contain pandas.Series
+            Data to transform
+        y : ignored argument for interface compatibility
+
+        Returns
+        -------
+        Xt : nested pandas DataFrame of shape [n_instances, n_features]
+            each cell of Xt contains pandas.Series
+            transformed version of X
+        """
+        index_list = X.index.get_level_values(X.index.names[-1])
+
+        with warnings.catch_warnings():
+            warnings.simplefilter(action="ignore", category=FutureWarning)
+            fnc = partial(self.aggregate_fn, **self.kwargs)
+            X_aggregate = getattr(X.rolling(window=self.subseq_len), "agg")(
+                fnc.func, **fnc.keywords
+            ).dropna()
+
+        indices = getattr(X_aggregate, f"idx{self.selector}")()
+
+        upper = pd.Categorical(indices, categories=index_list, ordered=True).codes + 1
+        lower = upper - self.subseq_len
+
+        dfs = [
+            X[col].iloc[l:u].reset_index(drop=True)
+            for col, l, u in zip(X.columns, lower, upper)
+        ]
+
+        return pd.concat(dfs, axis=1, ignore_index=False)
+
+    @classmethod
+    def get_test_params(cls, parameter_set="default"):
+        """Return testing parameter settings for the estimator.
+
+        Parameters
+        ----------
+        parameter_set : str, default="default"
+            Name of the set of test parameters to return, for use in tests. If no
+            special parameters are defined for a value, will return ``"default"`` set.
+
+
+        Returns
+        -------
+        params : dict or list of dict, default = {}
+            Parameters to create testing instances of the class
+            Each dict are parameters to construct an "interesting" test instance, i.e.,
+            ``MyClass(**params)`` or ``MyClass(**params[i])`` creates a valid test
+            instance.
+            ``create_test_instance`` uses the first (or only) dictionary in ``params``
+        """
+        params = [
```

Comment:
we should also add one param set that has only defaults, except for `subseq_len` which needs to be set.